sdk/core/src/macros.rs

```diff
@@ -26,6 +26,7 @@
 #[macro_export]
 macro_rules! setters {
     (@single $name:ident : $typ:ty => $transform:expr) => {
+        // TODO: Declare with_$name when https://github.com/rust-lang/rust/issues/29599 is fixed.
```

Comment:
The code as-is doesn't actually agree with the comment about it, that says using `setters!()` is equivalent to generating `with_$name` methods. That's not currently possible without taking a dependency on 1 of a couple crates out there that provide this, despite nightly toolset support since ~2016 (still not in stable). I'm capturing the tracking work to make the macro comment actually true, or we need to take a dependency (or duplicate) on those crates, or fix the comment so it doesn't say `with_$name` methods are generated, but that doesn't seem very idiomatic.

Reply:
Ah, I forgot I fixed the name. Still, excluding "with_" as a function prefix doesn't seem very idiomatic. I think we should change this to actually be `with_$name` before we GA if that's truly what's idiomatic. I'll open a tracking issue of our own.

Reply:
Opened https://github.com/Azure/azure-sdk-for-rust/issues/292. We do track issues this way in our other SDK repos as well.